src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -423,6 +423,25 @@ class ScenarioRepository final : public IScenarioRepository
         Scan(language);
 
         scenario_index_entry* scenario = GetByFilename(scenarioFileName);
+
+        // Check if this is an RCTC scenario that corresponds to a known RCT1/2 scenario or vice versa, see #12626
+        if (scenario == nullptr)
+        {
+            const std::string scenarioBaseName = String::ToStd(Path::GetFileNameWithoutExtension(scenarioFileName));
+            const std::string scenarioExtension = String::ToStd(Path::GetExtension(scenarioFileName));
+
+            if (String::Equals(scenarioExtension, ".sea", true))
+            {
+                // Get scenario using RCT2 style name of RCTC scenario
+                scenario = GetByFilename((scenarioBaseName + ".sc6").c_str());
+            }
+            else if (String::Equals(scenarioExtension, ".sc6", true))
+            {
+                // Get scenario using RCTC style name of RCT2 scenario
+                scenario = GetByFilename((scenarioBaseName + ".sea").c_str());
```

Comment:
These potentially recurse indefinitely if no matching scenario can be found.

Reply:
Hm, how exactly?

Reply:
Oh, my mistake. For some reason, I thought this was recursing from `GetByFilename`. Clearly, that is not the case. Sorry!